containers/unit_tests/TestUnorderedMap.hpp

```diff
@@ -441,17 +441,49 @@ TEST(TEST_CATEGORY, UnorderedMap_valid_empty) {
   ASSERT_TRUE(n.is_allocated());
 }
 
+/**
+ * This helper is needed because NVCC does not like extended lambdas
+ * in private member functions.
+ * Google Test bodies are private member functions. So it is incompatible.
+ * See also https://github.com/google/googletest/issues/4104.
+ */
+template <typename map_type>
+struct UnorderedMapInsert {
+  const map_type m_map;
+
+  //! Ensure shared ownership of @ref m_map.
+  UnorderedMapInsert(map_type map) : m_map(std::move(map)) {}
+
+  //! Insert a single value.
+  template <typename T>
+  void insert_single(const T& arg) const {
+    Kokkos::parallel_for(
+        Kokkos::RangePolicy<typename map_type::execution_space>(0, 1),
+        KOKKOS_CLASS_LAMBDA(const int ) { this->m_map.insert(arg); });
+  }
+
+  //! Insert multiple values.
+  template <typename... Args>
+  void insert(Args &&... args) const {
+    static_assert(sizeof...(Args) > 1, "Prefer the single value version");
+    constexpr size_t size = sizeof...(Args);
+    Kokkos::Array<typename map_type::key_type, size> values{
+        std::forward<Args>(args)...};
+    Kokkos::parallel_for(
+        Kokkos::RangePolicy<typename map_type::execution_space>(0, size),
+        KOKKOS_CLASS_LAMBDA(const int i) { this->m_map.insert(values[i]); });
+  }
+};
```

Comment:
Hi @dalg24 ! I couldn't use the same name for both functions and use something like `std::enable_if_t<(sizeof...(Args) > 1)>` to differentiate between the single and multiple values specializations, because Cuda does not like it. I think we have a similar situation in our code, and it works because we use `requires` instead. Do you know any tweak here or are you fine with one `insert` and one `insert_single`?